service/s3/s3crypto/s3crypto_client.go

```diff
@@ -0,0 +1,175 @@
+package s3crypto
+
+import (
+	"encoding/hex"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// Client is an S3 crypto client. By default we will use EncryptionOnly mode which
+// will use AES Wrap for key wrapping and AES GCM for content encryption.
+// AES GCM will load all data into memory. However, the rest of the content algorithms
+// do not load the entire contents into memory.
+type Client struct {
+	S3     *s3.S3
+	Config Config
+}
+
+// Config used to customize the Client
+type Config struct {
+	// Mode is used to dictate how we encrypt and decrypt the data.
+	//
+	// Defaults to AES GCM with the suggested pairing of AES Wrap/PRSA/KMS
+	Mode CryptoMode
+	// SaveStrategy will dictate where the envelope is saved.
+	//
+	// Defaults to the object's metadata
+	SaveStrategy
```

Comment:
Is `SaveStrategy` intended to be an anonymous field? Interfaces as anon fields can hide missing implementations.